tools/check-upcoming-maintenance.py

```diff
@@ -0,0 +1,130 @@
+import os
+import re
+import subprocess
+from typing import List
+from optparse import OptionParser
+
+class NodeMaintenance:
+    '''
+    Class to keep track of project, zones, and search terms for nodes, as well as results from gcloud queries
+    '''
+    def __init__(self, project, zones, regex):
+        self.project = project
+        res = subprocess.run("gcloud projects describe {}".format(self.project), shell=True, capture_output=True, text=True)
+        if res.returncode != 0:
+            raise Exception("{} does not exist or you may not have permission to access it".format(self.project))
+        
+        self.zones = None
+        if zones:
+            self.zones = zones.split(",")
+            for zone in self.zones:
+                res = subprocess.run("gcloud compute zones describe {}".format(zone), shell=True, capture_output=True, text=True)
+                if res.returncode != 0:
+                    raise Exception("{} does not exist".format(zone))
+        
+        self.regex = None
+        self.screen_reader = False
+        res = subprocess.run("gcloud config get accessibility/screen_reader", 
```

Comment:
nit: Although not technically wrong, I would avoid using the same var for multiple calls. You could use more descriptive variable names that reflect the specific context of the result (ex. screen_reader_res). IMO this makes code more readable and avoids mistakes where one result bleeds into another.

Reply:
OBE